.github/workflows/performance.yml

```diff
@@ -0,0 +1,160 @@
+  # - Print Performance tests results.
+  # - Update the environment to 6.1.1
+  # - Run Performance tests (6.1.1)
+  # - Print Performance tests results (6.1.1).
+  # - Set commit details.
+  # - Publish performance results.
+  # - Ensures version-controlled files are not modified or deleted.
+  performance:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
+
+    steps:
+      - name: Configure environment variables
+        run: |
+          echo "PHP_FPM_UID=$(id -u)" >> $GITHUB_ENV
+          echo "PHP_FPM_GID=$(id -g)" >> $GITHUB_ENV
+
+      - name: Checkout repository
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        with:
+          node-version-file: '.nvmrc'
+          cache: npm
+
+      - name: Log debug information
+        run: |
+          npm --version
+          node --version
+          curl --version
+          git --version
+          svn --version
+          locale -a
+
+      - name: Install npm Dependencies
+        run: npm ci
+
+      - name: Build WordPress
+        run: npm run build
+
+      - name: Start Docker environment
+        run: |
+          npm run env:start
+
+      - name: Log running Docker containers
+        run: docker ps -a
+
+      - name: Docker debug information
+        run: |
+          docker -v
+          docker-compose -v
+          docker-compose run --rm mysql mysql --version
+          docker-compose run --rm php php --version
+          docker-compose run --rm php php -m
+          docker-compose run --rm php php -i
+          docker-compose run --rm php locale -a
+
+      - name: Install WordPress
+        run: npm run env:install
+
+      - name: Install WordPress Importer plugin and import mock data
+        run: |
+          npm run env:cli -- plugin install wordpress-importer --activate
+          curl -O https://raw.githubusercontent.com/WPTT/theme-test-data/b9752e0533a5acbb876951a8cbb5bcc69a56474c/themeunittestdata.wordpress.xml
+          npm run env:cli -- import themeunittestdata.wordpress.xml --authors=create
+          rm themeunittestdata.wordpress.xml
+
+      - name: Update permalink structure
+        run: |
+          npm run env:cli -- rewrite structure '/%year%/%monthnum%/%postname%/'
+
+      - name: Install MU plugin
+        run: |
+          mkdir ./build/wp-content/mu-plugins
+          cp ./tests/performance/mu-plugins/server-timing.php ./build/wp-content/mu-plugins/server-timing.php
+
+      - name: Run Performance tests (current commit)
+        run: npm run test:performance
+
+      - name: Print Performance tests results
+        run: "node ./tests/performance/results.js"
+
+      - name: Update the environment to 6.1.1
+        run: |
+          npm run env:cli -- core update --version=6.1.1 --force --path=/var/www/build
```

Comment:
Thinking about this a bit more. If this runs in older branches, it wouldn't be as simple as grabbing the most recent point release. Say 6.3 is out and the workflow runs in the 6.2 branch. We wouldn't want to compare the performance difference to the latest `6.3.z` release.

I think I'd be OK leaving these 6.1.1 hard coded values for now in the interest of getting this up and running. As long as we circle back to it in the next few weeks and address this.

Reply:
The purpose that 6.1.1 serves in this workflow is to have a consistent version that serves as a baseline when running performance metrics on the current commit. Including a consistent baseline along with the commit that is being tested allows us to normalize variations in measurements caused by the GitHub runners. In the performance workflow in the Gutenberg repo, this gets updated only once for every major release. However, we're proposing that we leave this pinned to a consistent version, so our baseline stays constant over time, unless we decide to explicitly bump the baseline version.

Reply:
Thanks @desrosj for feedback. If we remove the `--version=6.1.1` then it will install latest version. For now it install `6.2-beta4-55454`

Reply:
Since the purpose of this step is to install the baseline version we're using, I think it's appropriate to leave it as is for now until we resolve whether want to control this value from an environment variable of some sort.

Reply:
Correct, we do want to have a hard-coded version here, it must not be always the current version. If we update this every time that a new WordPress release is out, it will skew all future numbers compared to the previous numbers. We can consider changing that in the future, but at least conceptually this needs to be a fixed version.

If we ever change what commit hash this is based on, we will have to acknowledge that performance data after this will not be 100% reliably comparable with the data that was recorded before that change.

Reply:
> Thanks @desrosj for feedback. If we remove the `--version=6.1.1` then it will install latest version. For now it install `6.2-beta4-55454`

Thanks for clarifying that @mukeshpanchal27! I just tested locally as well and I can confirm that is the case. When using the `update` command it will determine whether the site is currently running a development version or not before requesting the update. I tested `wp core download --force`, and that does seem to download the latest version regardless of whether the site is currently using a development version such as `trunk`/beta/rc. But, given the related points around _why_ we want to hard code `6.1.1` to have a consistent base, I think this suggestion can be disregarded.

> Since the purpose of this step is to install the baseline version we're using, I think it's appropriate to leave it as is for now until we resolve whether want to control this value from an environment variable of some sort.

That works for me. I didn't review any of the methodologies behind the metrics being used or reasoning behind what is being reported, just the Actions related code. So fine with leaving the version as 6.1.1 for now.

However, I would like to see this extracted to a workflow level environment variable up with `LOCAL_DIR` This way when we do decide to change the version being used as a baseline, it only requires the version to be changed in one location.

I think it would also be good to include a multi-line doc block above that variable definition explaining why that version is being used for future contributors/us. And there may also be value in adding a docblock at the very top above the workflow name briefly outlining the thinking behind the overall approach being used and why/what the benefit is.